### PR TITLE
Add new failure tests for multi-shard/CTE modify and cte coordinator pull

### PR DESCRIPTION
**Pull request by mtuncer**
Adds failure tests for:
- multi-shard update/delete
- modifying cte
- coordinator pull/push for cte

### CODE REVIEW

**Comment by metdos** [comment]
@mtuncer are we done with this PR?

File: src/test/regress/expected/failure_cte_subquery.out [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+(1 row)
+
+WITH cte_delete as (DELETE FROM users_table WHERE user_name in ('A', 'D') RETURNING *)
+INSERT INTO users_table SELECT * FROM cte_delete;
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+-- verify contents are the same
+SELECT citus.mitmproxy('conn.allow()');
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/test/regress/expected/failure_cte_subquery.out]: removed extra conn.allow()s it turned out there are many

File: src/test/regress/expected/failure_multi_shard_update_delete.out [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+  3 |  3
+(1 row)
+
+SELECT count(*) FILTER (WHERE b = 1) b1, count(*) FILTER (WHERE b = 2) AS b2 FROM t3;
+ b1 | b2 
+----+----
+  3 |  3
+(1 row)
+
+UPDATE t3 SET b = 1 WHERE b = 2 RETURNING *
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/test/regress/expected/failure_multi_shard_update_delete.out]: fixed

File: src/test/regress/expected/failure_multi_shard_update_delete.out [hunk 3]
```diff
@@ -0,0 +1,669 @@
+--
+-- failure_multi_shard_update_delete
+--
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/test/regress/expected/failure_multi_shard_update_delete.out]: @onderkalaci addressed all your comments.

**Pull request merged**